src/dynamic_boundary_conditions/tide/main_tide_slr.py

```diff
@@ -26,6 +29,163 @@
 log = logging.getLogger(__name__)
 
 
+class ValidationResult(NamedTuple):
+    """
+    Represents the result of checking validation on parameters.
+
+    Attributes
+    ----------
+    is_valid : bool
+        If True then the parameters are valid, if False then they are invalid.
+    invalid_reason : Optional[str]
+        An error message describing the reason the validation has failed. Can be `None` if `is_valid is True`.
+    """
+    is_valid: bool
+    invalid_reason: Optional[str]
+
+
+def validate_slr_parameters(
+    engine: Engine,
+    proj_year: int,
+    confidence_level: str,
+    ssp_scenario: str,
+    add_vlm: bool,
+    percentile: int,
+    increment_year: int = 1
+) -> ValidationResult:
+    """
+    Validate each of the sea-level-rise parameters have valid values by querying the database.
+    Returns a ValidationResult so that you can easily check for validation and the reason for failure.
+
+    Parameters
+    ----------
+    engine : Engine
+        The engine used to connect to the database.
+    proj_year : int
+        The projection year for which the combined tide and sea level rise data should be generated.
+    confidence_level : str
+        The desired confidence level for the sea level rise data. Valid values are 'low' or 'medium'.
+    ssp_scenario : str
+        The desired Shared Socioeconomic Pathways (SSP) scenario for the sea level rise data.
+        Valid options for both low and medium confidence are: 'SSP1-2.6', 'SSP2-4.5', or 'SSP5-8.5'.
+        Additional options for medium confidence are: 'SSP1-1.9' or 'SSP3-7.0'.
+    add_vlm : bool
+        Indicates whether Vertical Land Motion (VLM) should be included in the sea level rise data.
+        Set to True if VLM should be included, False otherwise.
+    percentile : int
+        The desired percentile for the sea level rise data. Valid values are 17, 50, or 83.
+    increment_year : int = 1
+        The year interval used for interpolating the sea level rise data. Defaults to 1 year.
+    Returns
```

Comment:
blank line in between 

```suggestion
        The year interval used for interpolating the sea level rise data. Defaults to 1 year.
    
    Returns
```